jvm/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -107,6 +107,11 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
     addTransformableTags(plan)
   }
 
+  private def checkIfStorageSupported(location: String): Boolean = {
+    // TODO(yuan): move this list to config
+    location.contains("hdfs:/") || location.contains("file:/")
+  }
+
```

Comment:
Each storage adapter have predefined test function, shouldn't somehow this be used? 

https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/storage_adapters/s3fs/S3Util.h
```c++
inline bool isS3File(const std::string_view filename)
```

Reply:
Can we move the logic to table scan's doValidate() interface? where velox can check if the data source is supported or not by itself?